packages/next/src/shared/lib/devtool/app-segment-tree.tsx

```diff
@@ -94,34 +168,39 @@ export function SegmentViewNode({
   pagePath: string
   children: ReactNode
 }) {
-  const devToolContext = useSegmentTreeContext()
+  const segmentTreeCtx = useContext(AppSegmentTreeContext)
```

Comment:
Why read this from context instead of from `window`?

Reply:
It was used to consume the parent page path to construct the relationship between child and parent, I used a trie to replace the react context implementation and it's not needed anymore.